lib/web/requirejs/domReady.js

```diff
@@ -89,7 +89,7 @@ define(function () {
         //so removing the || document.readyState === "interactive" test.
         //There is still a window.onload binding that should get fired if
         //DOMContentLoaded is missed.
-        if (document.readyState === "complete") {
+        if (document.readyState !== "loading") {
```

Comment:
Me and @ihor-sviziev tested both versions on our project.
Best performance shown when domReady.js is included into <head> section. However you need to modify it's code a bit - add package name to AMD definition:
```js
define("domReady", function(){ ... });
```

Benchmark results on real project with slow 3g mode enabled in Google Chrome dev console:
 - original - first load: 61 sec, second load: 10 sec
 - changes from this pull request - first load: 27 sec, second load: 6.7 sec
 - move domReady.js to `<head>` - first load: 9 sec, second load: 5 sec <== **best result**

Timing was measured by adding following code to `<head>`:
```js
require(["domReady"], function(domReady) {
	domReady(function() {
		console.log(window.performance.now());
	});	
});
```

Reply:
> There is an issue:
This condition automatically accepts any states which may be introduced in browsers in future.
For example: some browser adds state "initializing" and it breaks Magento2 UI, because it passes this check.

See my comment https://github.com/magento/magento2/pull/25527#issuecomment-554101719. Specifically,

> Luckily this isn't something we have to worry much about. This check is an extremely common pattern, and both the WHATWG and W3C have strong requirements to not break the web when adding new features. But it's also a fairly trivial change to update that logic in the other PR, if we did go with it.

The `!== 'loading'` pattern is already how jQuery's `$(document).ready(cb)` function works https://github.com/jquery/jquery/blob/05184cc448f4ed7715ddd6a5d724e167882415f1/src/core/ready.js#L66

Reply:
@DrewML FYI in the latest version of jQuery it has changed 
https://github.com/jquery/jquery/blob/3.5.1/src/core/ready.js#L67-L72